src/main/webapp/dev/js/common/instructorEnroll.js

```diff
@@ -0,0 +1,147 @@
+ *
+ * 2 by 5 spreadsheetData (after)
+ * "TestSection1|Team1||test1@xample.com|test1comments\n
+ *  TestSection2||TestName2|test2@example.com|\n"
+ *
+ * @returns {string} user data rows
+ */
+function getUserDataRows(spreadsheetData) {
+    // needs to check for '' as an initial empty row with null values will be converted to e.g. "||||" after .map
+    return spreadsheetData.filter(row => (!row.every(cell => cell === null || cell === '')))
+            .map(row => row.join('|'))
+            .join('\n');
+}
+
+/**
+ * Pushes data from spreadsheetDataRows into an array.
+ * Facilitates the function loadData for the Handsontable instance.
+ * @returns {Array} updated data
+ */
+function getUpdatedData(spreadsheetDataRows) {
+    return spreadsheetDataRows.map(row => row.split('|'));
+}
+
+/**
+ * Transforms the first uppercase letter of a string into a lowercase letter.
+ * @returns {string} string in all lowercase letters
+ */
+function unCapitalizeFirstLetter(string) {
+    return string.charAt(0).toLowerCase() + string.slice(1);
+}
+
+/**
+ * Converts returned AJAX data to a suitable format required by Handsontable.
+ * @returns {Array} student data
+ */
+function ajaxDataToHandsontableData(studentsData, handsontableColHeader) {
+    const headers = handsontableColHeader.map(unCapitalizeFirstLetter);
+    return studentsData.map(student => (headers.map(
+            header => student[header])));
+}
+
+/**
+ * Toggle the chevron image depending on the user's action.
+ */
+function toggleChevronImage(panelCollapse, toggleChevron) {
+    if ($(panelCollapse[0]).hasClass('checked')) { // panel is shown
+        $(toggleChevron).addClass('glyphicon-chevron-up').removeClass('glyphicon-chevron-down');
+    } else {
+        $(toggleChevron).addClass('glyphicon-chevron-down').removeClass('glyphicon-chevron-up');
+    }
+}
+
+/**
+ * Expands panel, showing the spreadsheet interface.
+ */
+function expandStudentsPanel(panelCollapse) {
+    $(panelCollapse[0]).collapse('show');
+    $(panelCollapse[0]).addClass('checked');
+}
+
+/**
+ * Collapses panel, hiding the spreadsheet interface.
+ */
+function collapseStudentsPanel(panelCollapse) {
+    $(panelCollapse[0]).collapse('hide');
+    $(panelCollapse[0]).removeClass('checked');
+}
+
+/**
+ * Expands/Collapses the panel depending on the current state of the panel.
+ */
+function toggleStudentsPanel($panelHeading, panelCollapse, displayIcon, toggleChevron) {
+    displayIcon.html('');
+    if ($(panelCollapse[0]).hasClass('checked')) { // panel is shown
+        collapseStudentsPanel(panelCollapse);
+    } else {
+        expandStudentsPanel(panelCollapse);
+    }
+    toggleChevronImage(panelCollapse, toggleChevron);
+}
+
+/**
+ * Displays a message informing the user that there are no existing students in the course.
+ */
+function displayNoExistingStudents(displayIcon) {
+    const statusMsg = `
+        <strong style="margin-left: 1em; margin-right: 1em;">
```

Comment:
Mixing style into javascript like this is a bad practice. Have the empty element ready and the appropriate CSS classes for different info, then update the class and textual content when necessary.